apps/nsqd/nsqd.go

```diff
@@ -88,6 +88,8 @@ func nsqFlagset() *flag.FlagSet {
 
 	authHTTPAddresses := app.StringArray{}
 	flagSet.Var(&authHTTPAddresses, "auth-http-address", "<addr>:<port> to query auth server (may be given multiple times)")
+	flagSet.Bool("http-auth-required", false, "require auth for HTTP clients")
+	flagSet.Bool("https-auth-required", false, "require auth for HTTPS clients")
```

Comment:
I've got to think about this some more, but I'm not excited about the idea of adding two more flags for this.

It might mean we need to rethink the existing combination of TLS/Auth related flags, choose good/flexible ones, and deprecate existing ones.  This might introduce some short term implementation pain but open up the door for getting this right going forward.